### PR TITLE
BUGFIX: Restore XLIFF translation fallback

### PR DESCRIPTION
**Pull request by fritjofbohm**
This modification fixes the translation fallback mechanism and the ability to read regionalized language variants.

Issues fixed:

* Translation fallback did not work. `Neos.Flow.i18n.fallbackRule.order` was ignored when looking for XLIFF files
* When using a regional `Neos.Flow.i18n.defaultLocale` (e.g. `de_AT` for an Austrian version of `de`), XLIFF files were never read from  a directory with the full `defaultLocale` name, but only from the base language directory. In the example, files in `Packages/Application/Vendor.Package/Resources/Private/Translations/de_AT` were never read, only from`Packages/Application/Vendor.Package/Resources/Private/Translations/de`.

### CODE REVIEW

**Comment by albe** [comment]
Thanks, this makes sense! Left two comments. Also, this should probably be targeted lower (3.3 even as it's LTS and still in bugfix support).

**Comment by albe** [comment]
Thanks! Can you also rebase this on 3.3, since simply changing the base won't work in this case.

**Comment by fritjofbohm** [comment]
This file is new in 4.2, so we can't rebase this to 3.3. Before 4.2, Flow used a different approach in loading XLIFF files (see https://github.com/neos/flow-development-collection/pull/894)

**Comment by albe** [comment]
Ah yes, of course! My bad :) Then this just needs a rebase on 4.2, which should be pretty easy.

File: Neos.Flow/Classes/I18n/Xliff/Service/XliffFileProvider.php [hunk 1]
```diff
@@ -87,26 +94,30 @@ public function initializeObject()
      */
     public function getMergedFileData($fileId, Locale $locale): array
     {
-        if (!isset($this->files[$fileId][$locale->getLanguage()])) {
```
Comment (aertmann) [inline Neos.Flow/Classes/I18n/Xliff/Service/XliffFileProvider.php]: hmm feels wrong to me to rely on the magic `__toString()` method, maybe add a `getIdentifier` or `getLocaleIdentifier` method and use that instead? `__toString()` could do that same then of course..

but maybe that's just me..

Comment (fritjofbohm) [reply Neos.Flow/Classes/I18n/Xliff/Service/XliffFileProvider.php]: I'd say a `getIdentifier()` method in `\Neos\Flow\I18n\Locale` is generally a good thing, but it's also a medium change to Flow if you consequently replace each `Locale` string cast with the new method. This is just a humble bugfix ;)

Comment (aertmann) [reply Neos.Flow/Classes/I18n/Xliff/Service/XliffFileProvider.php]: you're right, seems like it's used a bunch of other places in the code so guess someone meant to use that way.. nvm then

**Comment by fritjofbohm** [comment]
OK, I did a rebase of this stuff on 4.2. Please let me know if I did it right

**Comment by albe** [comment]
Seems a couple of unrelated commits sneaked in. Try pulling latest 4.2 branch, then (force) rebase on that and skip all but your two commits (if there are still any other).

**Comment by fritjofbohm** [comment]
Next try. Better this time?

**Pull request merged**